generate/generate.go

```diff
@@ -915,21 +915,46 @@ func (g *Generator) RemoveLinuxNamespace(ns string) error {
 }
 
 // AddDevice - add a device into g.spec.Linux.Devices
-func (g *Generator) AddDevice(path, devType string, major, minor int64, fileMode *os.FileMode, uid, gid *uint32) {
+func (g *Generator) AddDevice(device rspec.Device) {
 	g.initSpecLinux()
 
-	device := rspec.Device{
-		Path:     path,
-		Type:     devType,
-		Major:    major,
-		Minor:    minor,
-		FileMode: fileMode,
-		UID:      uid,
-		GID:      gid,
+	for i, dev := range g.spec.Linux.Devices {
+		if dev.Path == device.Path {
+			g.spec.Linux.Devices[i] = device
+			return
+		}
+		if dev.Type == device.Type && dev.Major == device.Major && dev.Minor == device.Minor {
+			fmt.Println("WARNING: The same type, major and minor should not be used for multiple devices.")
```

Comment:
I'm afraid this will break default output. Maybe fmt.Fprintln(os.Stderr, "") will be better